indico/modules/events/tracks/controllers.py

```diff
@@ -116,14 +141,41 @@ def _process(self):
         render_mode = track_settings.get(self.event, 'program_render_mode')
         program = RENDER_MODE_WRAPPER_MAP[render_mode](program)
         tracks = (Track.query.with_parent(self.event)
+                  .filter(~Track.track_group.has())
                   .options(subqueryload('conveners'),
                            subqueryload('abstract_reviewers'))
-                  .order_by(Track.position)
                   .all())
-        return WPDisplayTracks.render_template('display.html', self.event, program=program, tracks=tracks)
+        track_groups = self.event.track_groups
+        items = sorted(tracks + track_groups,  key=attrgetter('position'))
```

Comment:
I saw the same code in a few places, maybe we could add an Event's property which would return this kind of data?